BitwardenShared/Core/Platform/Utilities/ServerVersion.swift

```diff
@@ -0,0 +1,92 @@
+import Foundation
+import OSLog
+
+// MARK: - ServerVersion
+
+/// Model that represents the server version.
+///
+struct ServerVersion: Comparable, Codable {
+    // MARK: - Constants
+
+    /// The separator between version and a suffix..
+    private static let suffixSeparator: Character = "-"
+
+    /// The separator between version's components.
+    private static let versionSeparator: Character = "."
+
+    // MARK: Parameters
+
+    /// A string representing the version
+    var version: String
+
+    // MARK: Init
+
+    /// Initializes a `SeverVersion` with the `version` if it's valid, otherwise returns `nil`.
+    /// - Parameter version: The string`version` to use.
+    init?(_ version: String) {
+        let trimmedVersion = version.trimmingCharacters(in: .whitespacesAndNewlines)
+        guard Self.isValid(version: trimmedVersion) else {
+            return nil
+        }
+        self.version = trimmedVersion
+    }
+
+    // MARK: Methods
+
+    /// Compare if `lhs` is less than `rhs`
+    /// - Parameters:
+    ///   - lhs: Left hand side version to compare.
+    ///   - rhs: Right hand side version to compare.
+    /// - Returns: `true` if `lhs` is less than `rhs`, `false` otherwise.
+    static func < (lhs: ServerVersion, rhs: ServerVersion) -> Bool {
+        let lhsComponents = lhs.versionComponents()
+        let rhsComponents = rhs.versionComponents()
+
+        for (lhsComponent, rhsComponent) in zip(lhsComponents, rhsComponents) {
+            if lhsComponent < rhsComponent {
+                return true
+            } else if lhsComponent > rhsComponent {
+                return false
+            }
+        }
+        return false
+    }
+
+    /// Compare if `lhs` is equal to `rhs`
+    /// - Parameters:
+    ///   - lhs: Left hand side version to compare.
+    ///   - rhs: Right hand side version to compare.
+    /// - Returns: `true` if `lhs` is equal to `rhs`, `false` otherwise.
+    static func == (lhs: ServerVersion, rhs: ServerVersion) -> Bool {
+        lhs.version == rhs.version
+    }
+
+    /// Checks whether the `version` has the correct format.
+    /// - Parameter version: Version value to check.
+    /// - Returns: `true` if valid, `false` otherwise.
+    private static func isValid(version: String) -> Bool {
+        // Regular expression pattern for a version string: X.Y.Z-metadata
+        // where X, Y, Z are non-negative integers without leading zeros
+        // and X cannot be only zeros
+        // and an optional hyphen followed by alphanumeric metadata
+        let regexPattern = "^[1-9][0-9]{3}\\.[0-9]+\\.[0-9]+(-[a-zA-Z0-9]+)*$"
```

Comment:
🤔 This assumes that our major version number will always be a four-digit year, which is probably reasonable, but we could make the regex just `[1-9][0-9]*` in order to be slightly more flexible.

Reply:
Yeah, makes sense 👍